src/expression.js

```diff
@@ -648,8 +648,9 @@ pp.parseIdent = function(liberal) {
   let node = this.startNode()
   if (liberal && this.options.allowReserved == "never") liberal = false
   if (this.type === tt.name) {
-    if (!liberal &&
-        (this.strict ? this.reservedWordsStrict : this.reservedWords).test(this.value) &&
+    let reservedWords = this.options.sourceType === "module" ? this.reservedWordsES6Module :
```

Comment:
I was thinking about just modifying this line https://github.com/ternjs/acorn/blob/master/src/state.js#L20 depending on `this.inModule` which is defined a bit later in the constructor but can be easily moved up.